source/MaterialXRuntime/RtFileIo.cpp

```diff
@@ -793,7 +794,8 @@ namespace
             {
                 // Write input if it's connected or different from default value.
                 if (writeDefaultValues || 
-                    input.isConnected() || !RtValue::compare(input.getType(), input.getValue(), attrDef.getValue()))
+                    input.isConnected() || !RtValue::compare(input.getType(), input.getValue(), attrDef.getValue()) ||
+                    (input.getMetadata(UI_VISIBLE) && input.getMetadata(UI_VISIBLE)->getValueString() == "false"))
```

Comment:
I assume this is done to preserve the metadata? Maybe just add a code comment why the input is written out in this case (I misread the code the first time I saw it)

Reply:
So the goal is to write out the input if uivisible is a non-default value. I added a comment, let me know if that's sufficient.